daml-lf/spec/daml-lf-1.rst

```diff
@@ -1887,140 +1887,123 @@ will always be used to compare values of same types::
+    'Numeric' <ₜ 'ContractId'
 
-  ——————————————————————————————————————————————————— GenLtListNil
-   'Nil' @τ <ᵥ 'Cons' @σ wₜ wₜ
+  ———————————————————————————————————————————————————
+    'ContractId' <ₜ'Arrow'
 
-   vₕ <ᵥ wₕ
-  ——————————————————————————————————————————————————— GenLtListConsHead
-   'Cons' @τ vₕ vₜ  <ᵥ 'Cons' @σ wₜ wₜ
+  ———————————————————————————————————————————————————
+    'Arrow' <ₜ 'Optional'
 
-   vₕ ~ᵥ wₕ    vₜ <ᵥ wₜ
-  ——————————————————————————————————————————————————— GenLtListConsTail
-   'Cons' @τ vₕ vₜ  <ᵥ 'Cons' @σ wₜ wₜ
+  ———————————————————————————————————————————————————
+    'Optional' <ₜ 'List'
 
-  ——————————————————————————————————————————————————— GenLtOptionalNone
-   'None' @τ <ᵥ 'Some' @σ w
+  ———————————————————————————————————————————————————
+    'List' <ₜ 'TextMap'
 
-   v <ᵥ w
-  ——————————————————————————————————————————————————— GenLtOptionalSome
-   'Some' @τ v <ᵥ 'Some' @σ w
+  ———————————————————————————————————————————————————
+    'TextMap' <ₜ 'GenMap'
 
-   v₁ ~ᵥ v₁   ⋯   vᵢ₋₁ ~ᵥ wᵢ₋₁     vᵢ <ᵥ wᵢ    i <= n
-  ——————————————————————————————————————————————————— GenLtRecCon
-  Mod:T @τ1 … @τₙ { f₁ = v₁, …, fₙ = wₘ }
-    <ᵥ Mod:T @σ₁ … @σₙ { f₁ = w₁, …, fₙ = wₘ }
+  ———————————————————————————————————————————————————
+    'GenMap' <ₜ 'Any'
 
-   Mod:T:V₁ comes before Mod:T:V₂ in the list of
-   constructors for variant type Mod:T
-  ——————————————————————————————————————————————————— GenLtVariantCon1
-   Mod:T:V₁ @τ₁ … @τₙ v <ᵥ Mod:T:V₂ @σ₁ … @σₙ w
+  ———————————————————————————————————————————————————
+    'Any' <ₜ 'TypeRep'
 
-   v <ᵥ w
-  ——————————————————————————————————————————————————— GenLtVariantCon2
-   Mod:T:V @τ₁ … @τₙ v <ᵥ Mod:T:V @σ₁ … @σₙ w
+  ———————————————————————————————————————————————————
+    'TypeRep' <ₜ 'Update'
 
-   Mod:T:E₁ comes before Mod:T:E₂ in the list of
-   constructors for enum type Mod:T
-  ——————————————————————————————————————————————————— GenLtEnumCon
-   Mod:T:E₁ <ᵥ Mod:T:E₂
+  ———————————————————————————————————————————————————
+    'Update' <ₜ Mod:T
 
-   v₁ ~ᵥ v₁   ⋯   vᵢ₋₁ ~ᵥ wᵢ₋₁     vᵢ <ᵥ wᵢ    i <= n
-  ——————————————————————————————————————————————————— GenLtStructCon
-   ⟨ f₁ = v₁, …, fₘ = vₘ ⟩ ~ᵥ ⟨ f₁ = w₁, …, fₘ = wₘ ⟩
+    PkgId₁ < PkgId₂
+  ———————————————————————————————————————————————————
+    (PkgId₁:ModName₁):T₁ <ₜ (PkgId₂:ModName₂):T₂
+
+    ModName₁ < ModName₂
+  ———————————————————————————————————————————————————
+    (PkgId:ModName₁):T₁ <ₜ (PkgId:ModName₂):T₂
+
+    T₁ < T₂
+  ———————————————————————————————————————————————————
+    Mod:T₁ <ₜ Mod:T₂
+
+  ———————————————————————————————————————————————————
+    Mod:T <ₜ n
+
+    n₁ is strictly less that n₂
+  ———————————————————————————————————————————————————
+    n₁ <ₜ n₂
+
+  ———————————————————————————————————————————————————
+    n <ₜ ⟨ f₁ : τ₁, …, fₘ : τₘ ⟩
+
+    fᵢ < gᵢ
+  ———————————————————————————————————————————————————
+    ⟨ f₁ : τ₁, …, fₘ : τₘ ⟩ <ₜ
```

Comment:
Shouldn’t the types of the fields be the same, i.e, `t_i` everywhere?

Reply:
Struct may have different type for each field